src/app/content/highlights/components/CardWrapper.tsx

```diff
@@ -1,30 +1,121 @@
 import Highlighter, { Highlight } from '@openstax/highlighter';
+import { HTMLElement } from '@openstax/types/lib.dom';
 import React from 'react';
-import styled from 'styled-components/macro';
-import theme from '../../../theme';
+import { connect } from 'react-redux';
+import styled from 'styled-components';
+import { scrollIntoView } from '../../../domUtils';
+import { AppState } from '../../../types';
+import { assertDefined, assertNotNull, remsToPx } from '../../../utils';
+import * as selectSearch from '../../search/selectors';
+import * as contentSelect from '../../selectors';
+import { cardMarginBottom } from '../constants';
 import Card from './Card';
+import { mainWrapperStyles } from './cardStyles';
+import { getHighlightTopOffset } from './cardUtils';
 
-interface Props {
+export interface WrapperProps {
+  hasQuery: boolean;
+  isTocOpen: boolean;
   container: HTMLElement;
   highlighter: Highlighter;
   highlights: Highlight[];
   className?: string;
 }
 
 // tslint:disable-next-line:variable-name
-const Wrapper = ({highlights, className, container, highlighter}: Props) => highlights.length
-  ? <div className={className}>
-      {highlights.map((highlight) => <Card
+const Wrapper = ({highlights, className, container, highlighter}: WrapperProps) => {
+  const element = React.useRef<HTMLElement>(null);
+  const [cardsPositions, setCardsPositions] = React.useState<Map<string, number>>(new Map());
+  const [cardsHeights, setCardsHeights] = React.useState<Map<string, number>>(new Map());
+  const [topOffsets, setTopOffsets] = React.useState<Map<string, number>>(new Map());
+
+  const onHeightChange = (id: string, ref: React.RefObject<HTMLElement>) => {
+    const height = ref.current && ref.current.offsetHeight;
+    if (cardsHeights.get(id) !== height) {
+      setCardsHeights((previous) => new Map(previous).set(id, height === null ? 0 : height));
+    }
+  };
+
+  const onFocus = (highlight: Highlight) => {
+    const position = cardsPositions.get(highlight.id);
+    if (typeof position !== 'number') { return; }
+
+    const topOffset = getTopOffsetForHighlight(highlight);
+
+    if (position > topOffset) {
+      assertNotNull(element.current, 'element.current can\'t be null').style.top = `-${position - topOffset}px`;
+    }
```

Comment:
If it's possible, you should be animating with `transformY` instead of `top` since i'm pretty sure it doesn't make the browser repaint